internal/services/dedicatedcloud/dedicatedcloud.go

```diff
@@ -211,86 +220,27 @@ func ListDatacenter(_ *cobra.Command, args []string) {
 		for _, host := range hosts {
 			// Sum cores
 			if cpuNumRaw, ok := host["cpuNum"]; ok && cpuNumRaw != nil {
-				var cpuNumValue float64
-				switch v := cpuNumRaw.(type) {
-				case json.Number:
-					val, err := v.Float64()
-					if err == nil {
-						cpuNumValue = val
-					}
-				case float64:
-					cpuNumValue = v
-				case int:
-					cpuNumValue = float64(v)
-				case int64:
-					cpuNumValue = float64(v)
-				case int32:
-					cpuNumValue = float64(v)
-				}
-				totalCores += int(cpuNumValue)
+				totalCores += int(toFloat64(cpuNumRaw))
```

Comment:
```suggestion
				totalCores += toInt(cpuNumRaw)
```
or I miss something?